### PR TITLE
add new parser - terrascan

### PR DESCRIPTION
**Pull request by ansidorov**
Terrascan is a static code analyzer for Infrastructure as Code.

It could be found at: https://github.com/accurics/terrascan

### CODE REVIEW

**Review by valentijnscholten — APPROVED** [review]
Can you check if we need to set dedupe algorithm to HASH_CODE in settings? And if we need a hash code field config or can use the default? (Might be good to check the same for the previous parsers by you which we already merged)

**Comment by ansidorov** [comment]
> Can you check if we need to set dedupe algorithm to HASH_CODE in settings? And if we need a hash code field config or can use the default? (Might be good to check the same for the previous parsers by you which we already merged)

If I understand correctly, we need to use the default for terrascan parser.

**Review by devGregA — APPROVED** [review]

**Comment by ansidorov** [comment]
> Can you check if we need to set dedupe algorithm to HASH_CODE in settings? And if we need a hash code field config or can use the default? (Might be good to check the same for the previous parsers by you which we already merged)

I thought again and adjusted the deduplication algorithm.

File: dojo/tools/terrascan/parser.py [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+        return "Terrascan Scan"
+
+    def get_description_for_scan_types(self, scan_type):
+        return "Import JSON output for Terrascan scan report."
+
+    def get_findings(self, filename, test):
+        data = json.load(filename)
+        dupes = {}
+        if data.get('results').get('violations'):
+            for item in data.get('results').get('violations'):
```
Comment (damiencarol) [inline dojo/tools/terrascan/parser.py:31]: This conditions can fail if `results` attribute is not here.
```suggestion
        if 'results' not in data and 'violations' not in data.get('results'):
            raise ValueError("missing mandatory attribute 'results'")
        for item in data.get('results').get('violations'):
```

File: dojo/tools/terrascan/parser.py [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+
+    def get_description_for_scan_types(self, scan_type):
+        return "Import JSON output for Terrascan scan report."
+
+    def get_findings(self, filename, test):
+        data = json.load(filename)
+        dupes = {}
+        if 'results' not in data and 'violations' not in data.get('results'):
+            raise ValueError("missing mandatory attribute 'results'")
+        for item in data.get('results').get('violations'):
```
Comment (damiencarol) [inline dojo/tools/terrascan/parser.py:32]:
```suggestion
        if data.get('results') is None:
            return list()
        for item in data.get('results').get('violations'):
```

**Review by damiencarol — APPROVED** [review]

**Pull request merged**